### PR TITLE
Bulk Strategy Update and Update in GET_ALL parameter of GET method

### PR DESCRIPTION
**Pull request by jitendra-xoriant**
Added Support to Bulk Strategy update
Modified GET_ALL parameter of GET function to return all records.

### CODE REVIEW

File: src/main/java/com/mediamath/terminalone/TerminalOne.java [hunk 1]
```diff
@@ -596,17 +648,102 @@ public VideoCreativeUploadStatus getVideoCreativeUploadStatus(String creativeId)
...
+	private JsonArray extractData(String response) {
+		JsonParser parser = new JsonParser();
+		JsonObject obj = parser.parse(response).getAsJsonObject();
+		return obj.get("data").getAsJsonArray();
+	}
+	
+	
+	public List<Strategy> bulkUpdate(List<Strategy> strategyList, Strategy updator) throws CloneNotSupportedException, ClientException, ParseException{
+		
+		List<Strategy> UpdatorStrategyList = new ArrayList<Strategy>();
```
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/TerminalOne.java]: I was thinking something like this. For example, if client wants to only update goalValue then they create a the following update model,
`Strategy s = new Strategy();
s.setGoalValue(2.0);
`
and pass to our method with list of strategies.

We need to then parse this update object to get non-null and non-false values (to tackle boolean default). We can use some reflection methods to do this.

`for (Method method : someObject.getClass().getDeclaredMethods()) {
    if (Modifier.isPublic(method.getModifiers())
        && method.getParameterTypes().length == 0
        && method.getReturnType() != void.class
        && (method.getName().startsWith("get") || method.getName().startsWith("is"))
    ) {
        Object value = method.invoke(someObject);
        if (value != null) {
            System.out.println(method.getName() + "=" + value);
        }
    }
}`

Then only update these non-null or non-false fields in the strategy list. This way we need not worry about setting compulsory fields.

File: src/main/java/com/mediamath/terminalone/TerminalOne.java [hunk 2]
```diff
@@ -596,17 +648,102 @@ public VideoCreativeUploadStatus getVideoCreativeUploadStatus(String creativeId)
...
+				// parse the data to entities.
+				try {
+					pageJsonResponse1 = parseGetData(pageResponse, query);
+				} catch (ParseException parseException) {
+					throw new ClientException("Unable to parse the response");
+				}
+				
+				jsonResponse = pageJsonResponse1;
+				lastCallResponse=pageResponse;
+			}
```
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/TerminalOne.java]: looks good.

File: src/main/java/com/mediamath/terminalone/TerminalOne.java (hunk 1 above)
Comment (jitendra-xoriant) [inline src/main/java/com/mediamath/terminalone/TerminalOne.java]: let me try this, also for this, I have to work on how to set values in other object.
Some doubts
1. What if we want to set boolean value to false?
2. What about the primitive values?

Comment (jitendra-xoriant) [inline src/main/java/com/mediamath/terminalone/TerminalOne.java]: Updated code with reflection based entity update.

**Pull request merged**